### PR TITLE
Refactor Pool View

### PR DESCRIPTION
**Pull request by Denchick**

### CODE REVIEW

**Review by rkhapov — APPROVED** [review]

File: pkg/pool/pool.go [hunk 1]
```diff
@@ -55,8 +57,6 @@ type PoolIterator interface {
 type ConnectionAllocFn func(shardKey kr.ShardKey, host string, rule *config.BackendRule) (shard.Shard, error)
 
 type DBPool interface {
-	shard.ShardIterator
-	PoolIterator
 	MultiShardPool
```
Comment (Denchick) [inline pkg/pool/pool.go:60]: MultiShardPool includes ShardIterator and PoolIterator

File: pkg/pool/dummypool.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import (
+	"sync"
+
+	"github.com/pg-sharding/spqr/pkg/models/kr"
+	"github.com/pg-sharding/spqr/pkg/models/spqrerror"
+	protos "github.com/pg-sharding/spqr/pkg/protos"
+	"github.com/pg-sharding/spqr/pkg/shard"
+)
+
+type DummyPool struct {
```
Comment (reshke) [inline pkg/pool/dummypool.go:12]: I do dislike Dummm* naming. I do not understand why ConnectionKeeper is bad.

File: pkg/pool/dummypool.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	Id            string
+	DB            string
+	Usr           string
+	Host          string
+	Router        string
+	ConnCount     int64
+	IdleConnCount int64
+	QueueSize     int64
```
Comment (reshke) [inline pkg/pool/dummypool.go:20]:
```
func (r *ConnectionKepperData) IdleConnectionCount() int {
	r.m.Lock()
	defer r.m.Unlock()

	return int(r.IdleConnCount)
}
```

this is how it used to look like. Where is this code? why do we rename everything and drop functionality?

File: pkg/pool/dummypool.go (hunk 2 above)
Comment (Denchick) [reply pkg/pool/dummypool.go:12]: ConnectionKeeper is a good naming, I combine CoordPool and ConnectionKeeperData here

**Review by reshke — APPROVED** [review]

**Pull request merged**